xdsl/transforms/scf_parallel_loop_tiling.py

```diff
@@ -0,0 +1,156 @@
+from dataclasses import dataclass
+
+from xdsl.dialects import affine, arith
+from xdsl.dialects.builtin import IndexType, IntegerAttr, ModuleOp
+from xdsl.dialects.scf import ParallelOp, Yield
+from xdsl.ir import Block, MLContext, Operation, Region, SSAValue
+from xdsl.passes import ModulePass
+from xdsl.pattern_rewriter import (
+    GreedyRewritePatternApplier,
+    PatternRewriter,
+    PatternRewriteWalker,
+    RewritePattern,
+    op_type_rewrite_pattern,
+)
+from xdsl.utils.hints import isa
+
+
+@dataclass
+class ScfParallelLoopTilingPattern(RewritePattern):
+    tile_sizes: list[int]
+
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: ParallelOp, rewriter: PatternRewriter, /):
+        # Only tile the innermost parallel loop
+        if any(isinstance(o, ParallelOp) for o in op.body.walk()):
+            return
+        lower = op.lowerBound
+        upper = op.upperBound
+        step = op.step
+
+        # fill the tile sizes with ones
+        tile_sizes_v = self.tile_sizes
```

Comment:
this is a little scary, why the mutable state?

Reply:
I'm filling it with ones if needed, which is safe for this specific pass logic (Either I had enough values, either I'd add ones anyway)
Now that you mention it, I get the cringe though. Will switch to a copy for safer future changes